src/components/Voting/VotingCard.tsx

```diff
@@ -4,15 +4,32 @@ import RemoveMarkdown from 'remove-markdown'
 import Card from 'components/Card'
 import FormattedDate from 'components/Voting/FormattedDate'
 
-import { VotingProposalProps } from 'models/voting'
+import {
+  VotingChoices,
+  VotingProposalProps,
+  VotingOutcomes,
+  VotingStatus,
+} from 'models/voting'
 
 interface VotingCardProps {
   proposal: VotingProposalProps
 }
 
 const VotingCard: React.FC<VotingCardProps> = ({
-  proposal: { state, title, body, start, id },
+  proposal: { state, title, body, start, id, choices, scores },
 }) => {
+  let stateText = state
+  if (state === VotingStatus.CLOSED) {
+    const highestVotedIndex = scores?.indexOf(Math.max(...scores))
+    if (choices[highestVotedIndex] === VotingChoices.FOR) {
+      stateText = VotingOutcomes.PASSED
+    } else if (choices[highestVotedIndex] === VotingChoices.AGAINST) {
+      stateText = VotingOutcomes.FAILED
+    } else {
+      stateText = VotingOutcomes.ABSTAINED
+    }
+  }
```

Comment:
hey @mitesh-mutha i feel like these lines can be improved and follow the common patterns of react.

```suggestion
  const [stateText, setStateText] = useState(state)
  
  useEffect(() => {
    if (state === VotingStatus.CLOSED) {
      const highestVotedIndex = scores?.indexOf(Math.max(...scores))
      if (choices[highestVotedIndex] === VotingChoices.FOR) {
        setStateText(VotingOutcomes.PASSED)
      } else if (choices[highestVotedIndex] === VotingChoices.AGAINST) {
        setStateText(VotingOutcomes.FAILED)
      } else {
        setStateText(VotingOutcomes.ABSTAINED)
      }
    }
  
  }, [state])
  

```

Reply:
@arjanvillon Thanks for the review! Noted. Will incorporate the changes suggested.